client/src/main/java/org/eclipse/hono/client/impl/CredentialsClientFactoryImpl.java

```diff
@@ -67,17 +67,18 @@ protected void onDisconnect() {
             final String tenantId) {
 
         Objects.requireNonNull(tenantId);
-        return connection.executeOnContext(result -> {
-            credentialsClientFactory.getOrCreateClient(
-                    CredentialsClientImpl.getTargetAddress(tenantId),
-                    () -> CredentialsClientImpl.create(
-                            cacheProvider,
-                            connection,
-                            tenantId,
-                            this::removeCredentialsClient,
-                            this::removeCredentialsClient),
-                    result);
-        });
+        return connection.isConnected(connection.getConfig().getRequestTimeout())
```

Comment:
in all the other places we use the request timeout for limiting the time to wait for the processing of a request over an already established connection, don't we? So, in order to improve chances that the request, that his method invocation is part of, succeeds, we might want to slightly reduce the time to wait a little so that there is some time left to establish the link and send the request message?

Reply:
That would only be needed if we had another timeout mechanism around the combind "getOrCreateClient+sendRequest" invocations, with a timeout value of requestTimeout+X.

But I don't see such a timeout mechanism being in place and therefore the number of milliseconds to subtract here would just be arbitrary. And then I would prefer to keep using the full request timeout value.

And thinking of the default value of 200ms here, I wouldn't want to decrease that value further (I think a somewhat higher value would even be better in many cases).

Reply:
You are right in pointing out that the request timeout only covers the time it takes to send the request and receive the response over an already established link. In fact, when a client invokes this method and the connection is established but the link for the given tenant is not established, then the client will wait for at most `ClientConfigProperties.linkEstablishmentTimeout` milis for the link to be established before timing out.

So, if the connection is *not* established but an attempt to connect is already going on, I guess that if we allow for roughly the same amount of time to establish the connection, the client should get a fair chance to get its request fulfilled but should not be required to wait exceptionally long for a failed result. WDYT?

Reply:
Yes, I think that would be a good choice. I've adapted the commit accordingly.